build.gradle

```diff
@@ -167,22 +167,27 @@ eclipseClasspath {
     }
 }
 
+def isTravis = System.getenv("TRAVIS") != null
+def isAppVeyor = System.getenv("APPVEYOR") != null
+def isCi = isTravis || isAppVeyor
+
 task createConfigs {
     description "Sets up the project by obtaining necessary files and configurations points."
     group "Setup"
+    def testTemplateFilename = isCi ? "test.ci.properties" : "test.template.properties"
```

Comment:
I'd like to disagree with this.

> Travis should use a different unique ID from the default so that hardcoding can be detected

With this PR, all one needs to do is using "TravisCi" as the unique ID and the problem will still persist.
I'd suggest, with the improvement added in #7379, to just override the IDs if God mode is used or it is a CI environment. You could do something like:
```java
...
TEST_ADMIN_ACCOUNT = randomizeUsername(prop.getProperty("test.admin.account"), "yourGoogleId");
// do the same for all accounts
...
private static String randomizeUsername(String username, String defaultUsername) {
    String randomId = generateRandomSequence(); // implement accordingly
    return isCiEnvironment() || IS_GOD_MODE_ENABLED ? defaultUsername + "." + randomId : username;
}
```

The above can achieve true randomness, and removes the need for are-all-test-accounts-ready check. You may argue that `yourGoogleId`, `teammates.coord`, `alice.tmms`, `charlie.tmms` are all hard-coded, but they have been from the beginning.

> Also, we can tweak this file to fine-tune CI performance (e.g. optimal timeout value specific to the CI environment that leads to faster builds yet minimises false positives), independent of the default values we'd like all developers to start with.

The moment a dev has his/her own `test.properties`, whatever value designated at the template file has no relevance anymore. You could even say that `test.template.properties` is pre-occupied with values to work in CI environment (whether they actually are currently is another issue).
Besides, is it not the case that the values that are guaranteed to work in CI environment are also the values we'd like all developers to start with?